models/avatars/avatar.go

```diff
@@ -109,39 +106,69 @@ func HashedAvatarLink(email string, size int) string {
 		}
 		return lowerEmail, nil
 	})
-	if size > 0 {
-		return setting.AppSubURL + "/avatar/" + url.PathEscape(sum) + "?size=" + strconv.Itoa(size)
-	}
-	return setting.AppSubURL + "/avatar/" + url.PathEscape(sum)
+	return emailHash
 }
 
-// MakeFinalAvatarURL constructs the final avatar URL string
-func MakeFinalAvatarURL(u *url.URL, size int) string {
-	vals := u.Query()
-	vals.Set("d", "identicon")
-	if size != DefaultAvatarSize {
-		vals.Set("s", strconv.Itoa(size))
+// GenerateUserAvatarFastLink returns a fast link to the user's avatar via the local explore page.
+func GenerateUserAvatarFastLink(userName string, size int) string {
+	if size < 0 {
+		size = 0
 	}
-	u.RawQuery = vals.Encode()
-	return u.String()
+	link := setting.AppSubURL + "/user/avatar/" + userName + "/" + strconv.Itoa(size)
+	return setting.AppURL + strings.TrimPrefix(link, setting.AppSubURL)[1:]
```

Comment:
Can't it be written as the following, or am I misunderstanding something here?
```suggestion
	return setting.AppURL + "/user/avatar/" + userName + "/" + strconv.Itoa(size)
```
Alternatively, if I interpret the `[1:]` correctly:
```suggestion
	return setting.AppURL + "user/avatar/" + userName + "/" + strconv.Itoa(size)
```
Alternatively, the version using formatted strings would be possible here but that would require the re-import of `fmt`.

Reply:
They are from the legacy code. I do not want to change the logic now.

```
func (u *User) SizedRelAvatarLink(size int) string {
	return setting.AppSubURL + "/user/avatar/" + u.Name + "/" + strconv.Itoa(size)
}
func (u *User) RelAvatarLink() string {
	return u.SizedRelAvatarLink(DefaultAvatarSize)
}
func (u *User) AvatarLink() string {
	link := u.RelAvatarLink()
	if link[0] == '/' && link[1] != '/' {
		return setting.AppURL + strings.TrimPrefix(link, setting.AppSubURL)[1:]
	}
	return link
}
```

Reply:
@delvh just searched a while, I think we use `setting.AppSubURL + "/user/avatar/" + userName + "/" + strconv.Itoa(size)` can be better here. I am not sure about `should we use AppURL or StaticURLPrefix`. Anyway, it just works as before now.